pilot/pkg/model/jwks_resolver.go

```diff
@@ -310,34 +341,34 @@ func (r *jwksResolver) refresh(t time.Time) {
 
 		oldPubKey := e.pubKey
 
-		// key rotation: fetch JWT public key again if it's expired.
-		if e.expireTime.Before(t) {
```

Comment:
why remove this line ?  should re-fetch when key expire

Reply:
The direct reason is it's the result of increasing the public key expiration duration, see my reply in the other comment.

The other reason is it looks no need to make the refresh job to depend on the public key expiration duration, it seems better to let the refresh job to update the public key independently.